internal/docker_test.go

```diff
@@ -0,0 +1,480 @@
+
+			Convey("Get the container id of a new container", func() {
+				// For client with empty list of containers
+				id, err := empDockerClient.GetNewDockerContainerID()
+				So(err, ShouldNotBeNil)
+				So(id, ShouldBeEmpty)
+
+				// For client with list of containers with container_id4 considered as a new container
+				id, err = dockerClient.GetNewDockerContainerID()
+				So(err, ShouldBeNil)
+				So(id, ShouldEqual, "container_id4")
+
+				// Mark the new container as seen now
+				err = dockerClient.RememberCurrentContainerIDs()
+				So(err, ShouldBeNil)
+
+				id, err = dockerClient.GetNewDockerContainerID()
+				So(err, ShouldNotBeNil)
+				So(id, ShouldBeEmpty)
+			})
+
+			Convey("Verify a container id from a container list", func() {
+				// For client with empty list of containers
+				boolOut, err := empDockerClient.verifyID("wrong_id")
+				So(boolOut, ShouldBeFalse)
+				So(err, ShouldNotBeNil)
+
+				// For client with list of containers
+				boolOut, err = dockerClient.verifyID("wrong_id")
+				So(boolOut, ShouldBeFalse)
+				So(err, ShouldNotBeNil)
+
+				boolOut, err = dockerClient.verifyID("container_id1")
+				So(boolOut, ShouldBeTrue)
+				So(err, ShouldBeNil)
+			})
+
+			Convey("Get container id given a name from its list of names", func() {
+				container4 := clist[3]
+				name, err := getIDByName("wrong_name", container4)
+				So(name, ShouldBeEmpty)
+				So(err, ShouldNotBeNil)
+
+				name, err = getIDByName("test_container4_new", container4)
+				So(name, ShouldEqual, "container_id4")
+				So(err, ShouldBeNil)
+			})
+
+			Convey("Get new container's id given a name from a container list", func() {
+				// For client with empty list of containers
+				name, err := empDockerClient.GetNewDockerContainerIDByName("wrong_name")
+				So(name, ShouldBeEmpty)
+				So(err, ShouldNotBeNil)
+
+				// For client with list of containers
+				name, err = dockerClient.GetNewDockerContainerIDByName("wrong_name")
+				So(name, ShouldBeEmpty)
+				So(err, ShouldNotBeNil)
+
+				// New Container
+				name, err = dockerClient.GetNewDockerContainerIDByName("test_container4")
+				So(name, ShouldEqual, "container_id4")
+				So(err, ShouldBeNil)
+
+				// Remember the new container and check it gets added to exisiting containers
+				err = dockerClient.RememberCurrentContainerIDs()
+				So(err, ShouldBeNil)
+				So(dockerClient.existingContainers["container_id4"], ShouldBeTrue)
+
+				// Get id if there is no new container
+				name, err = dockerClient.GetNewDockerContainerIDByName("test_container1")
+				So(name, ShouldBeEmpty)
+				So(err, ShouldNotBeNil)
+
+				// Mark container_id4 marked as false in exisiting container, making it "new" container
+				dockerClient.existingContainers["container_id4"] = false
+				So(dockerClient.existingContainers["container_id4"], ShouldBeFalse)
+			})
+
+			Convey("Given a file path read and return container id from a file", func() {
+				// cidPathToID
```

Comment:
Move the file creation code in to here and nest Conveys for the different methods, instead of using comments

Reply:
Done